components/net/cookie_storage.rs

```diff
@@ -78,57 +75,18 @@ impl CookieStorage {
         }
 
         // Step 12
-        let domain = reg_host(&cookie.cookie.domain.clone().unwrap_or("".to_owned())).unwrap_or("".to_owned());
-        if !self.cookies_map.contains_key(&domain) {
-        let tempdomain = domain.clone();
-            let cookies: Vec<Cookie> = Vec::new();
-            self.cookies_map.insert(tempdomain, cookies);
-        }
-        let mut cookies = self.cookies_map.get_mut(&domain).unwrap();
+        let domain = reg_host(&cookie.cookie.domain.as_ref().unwrap_or(&"".to_string()));
```

Comment:
We got the error when we used `unwrap_or("")` expected String reference found str. So we used `&"".to_string()`